webpack/assets/javascripts/react_app/ReactApp/ReactApp.js

```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Router } from 'react-router-dom';
+import { isEmpty } from 'lodash';
+import history from '../history';
+
+import Layout from '../components/Layout';
+import AppSwitcher from '../routes';
+
+const ReactApp = ({
+  data: { layout, metadata },
+  initializeMetadata,
+  stateMetadata,
+}) => {
+  if (isEmpty(stateMetadata)) initializeMetadata(metadata);
```

Comment:
Shouldn't it happen only `onMount`?

Reply:
this is to prevent the action if not needed?